first_assignment.py

```diff
@@ -1,11 +1,91 @@
-from PyQt5.QtWidgets import QApplication, QMainWindow, QWidget, QVBoxLayout, QLineEdit
-from PyQt5.QtGui import QPalette, QColor
+from PyQt5.QtWidgets import (
+    QApplication, QMainWindow, QLabel, QPushButton, 
+    QCheckBox, QSlider, QComboBox, QWidget, 
+    QHBoxLayout, QVBoxLayout, QLineEdit
+)
+from PyQt5.QtCore import Qt
 
 class MainWindow(QMainWindow):
+    """Custom window meant to display a textbox, dropdown bar,
+    yes/no checkboxes, slider from 0 to 100, and save button when
+    viewed from top to bottom.
+    """
     def __init__(self):
+        """Window constructor."""
+        # Declare instance variables required for YAML file
+        self.text = ""
+        self.bool = None
+        self.choice = ""
+        self.slider_value = 0
+
         super().__init__()
 
         self.setWindowTitle("Example GUI")
 
-        layout = QVBoxLayout()
-        layout.addWidget(QLineEdit())
\ No newline at end of file
+        # Set up overarching layouts
+        outer_layout = QVBoxLayout()
+        inner_layout = QHBoxLayout()    # will hold yes/no checkboxes
+
+        # Add yes/no checkboxes to the same row
+        self.yes_box = QCheckBox("Yes")
```

Comment:
I guess you only need one checkbox, but I like what you did with the functions that make it possible to select only one option, nice.